include/CLI/Validators.hpp

```diff
@@ -32,37 +32,75 @@ class Option;
 
 ///
 class Validator {
-    friend Option;
-
   protected:
-    /// This is the type name, if empty the type name will not be changed
-    std::string tname;
+    /// This is the validator description
+    std::string description_;
 
-    /// This is the type function, if empty the tname will be used
-    std::function<std::string()> tname_function;
+    /// This is the description function, if empty the description_ will be used
+    std::function<std::string()> desc_function;
 
     /// This it the base function that is to be called.
     /// Returns a string error message if validation fails.
     std::function<std::string(std::string &)> func;
+    /// The name for search purposes of the Validator
+    std::string name_;
+    /// Enable for Validator to allow it to be disabled if need be
+    bool active_{true};
 
   public:
+    Validator() = default;
+    // Construct Validator from basic information
+    explicit Validator(std::function<std::string(std::string &)> op, std::string name = "", std::string desc = "") {
```

Comment:
```suggestion
    explicit Validator(std::function<std::string(std::string &)> op, std::string name = "", std::string desc = "")
        : func(std::move(op)), name_(std::move(name)), description_(std::move(desc)): {}
```

(Note: needs to be formatted correctly)

Reply:
fixed